internal/common/maps/maps.go

```diff
@@ -37,3 +39,22 @@ func CloneStringMap(m map[string]string) map[string]string {
 	}
 	return m2
 }
+
+func OrderMapByKey[K any](input map[string]K) map[string]K {
```

Comment:
What property of the map does this affect? From what I can tell, [map iteration order isn't stable](https://go.dev/blog/maps#iteration-order), so changing the insertion order shouldn't have an impact on iteration order.

Reply:
hmm good point, do you think we should throw out this PR then?